python/ray/data/_internal/arrow_ops/transform_pyarrow.py

```diff
@@ -54,18 +54,28 @@ def unify_schemas(
         ArrowTensorType,
         ArrowVariableShapedTensorType,
     )
+    import pyarrow as pa
 
     schemas_to_unify = []
     schema_tensor_field_overrides = {}
```

Comment:
Could we rename this to `schema_field_overrides`, now that it includes both tensor overrides and null-typed list overrides?